### PR TITLE
feat(components-forms): Enumeration - add lazy-loading

### PR DESCRIPTION
**Pull request by Sylvain-Talend**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
- no callback when scrolling to the bottom

**What is the new behavior?**
- callback when scrolling to the bottom => possible to lazy-load

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:
Spinner will be added in other PR

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

File: packages/components/src/Enumeration/Items/Items.component.js [hunk 1]
```diff
@@ -65,8 +65,14 @@ function Items({ items, itemsProp, currentEdit, searchCriteria }) {
 		}
 	}
 
+	function onScroll(event) {
+		if (event.target.scrollTop + event.target.clientHeight >= event.target.scrollHeight) {
```
Comment (ncomont) [inline packages/components/src/Enumeration/Items/Items.component.js]: scrollTop, clientHeight, scrollHeight are called on every scroll event (around 400 times for 50 elements, see storybook example). These methods are time consuming. Could you "debounce" some of these calls ?

Comment (Sylvain-Talend) [inline packages/components/src/Enumeration/Items/Items.component.js]: Good point, I fix it

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
question: lazy loading is managed by virtualized.
AFAIK there is a topic on that for Enumeration to use it.

**Comment by Sylvain-Talend** [comment]
@jmfrancois PR on virtualized (#332) is different from this one, as in this one, we are able to load chunks of data with the new callback. So The REST request is done in the host application, and we can fully control it (size of the chunk, etc.)

**Comment by jmfrancois** [comment]
But I mean you can do lazy loading with virtualized on the server side to control the requests

https://bvaughn.github.io/react-virtualized/#/components/InfiniteLoader ?

_loadMoreRows ({ startIndex, stopIndex })

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by Sylvain-Talend** [comment]
@jmfrancois ok I look at that => WIP

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by Sylvain-Talend** [comment]
@jmfrancois as discussed, adding Rest calls into `loadMoreRows` will fire too much HTTP requests. For example, if chunk of 100 items and we have 100 000 items => 1000 calls will be resulting by using _infiniteLoader_ functionnality of react-virtualized

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://348.talend.surge.sh)

**Pull request merged**